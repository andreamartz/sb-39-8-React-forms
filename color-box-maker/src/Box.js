/**
 * Box component:
 * 
 *  * displays a div with a background color, width and height based on the props passed to it.
 */

import React from "react";

const Box = ({ id, key, width, height, backgroundColor, removeBox }) => {
  const handleRemove = () => removeBox(id);
  return (
    <div>
      <div
        style={{
          display: "inline-block",
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor
        }}
        aria-label={`${backgroundColor} box`}
      >
      </div>
      <button aria-label='Remove box' onClick={handleRemove}>X</button>
    </div>
  )
}

export default Box;