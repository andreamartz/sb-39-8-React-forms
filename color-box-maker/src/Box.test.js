import React from "react";
import { render } from '@testing-library/react';
import Box from './Box';

describe("Smoke tests", () => {
  test('renders without crashing', () => {
    render(<Box />);
  });
});

describe("Snapshot tests", () => {
  test('matches snapshot', () => {
    const {asFragment} = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
  });
});