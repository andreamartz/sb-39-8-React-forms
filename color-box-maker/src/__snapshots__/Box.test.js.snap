// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      aria-label="undefined box"
      style="display: inline-block;"
    />
    <button
      aria-label="Remove box"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
